annotation-utils/src/main/java/edu/isi/vista/annotationutils/PushAnnotations.kt

```diff
@@ -0,0 +1,177 @@
+        restoreJsonParamsBuilder.set("restoredJsonDirectory", params.getString("restoredJsonDirectory"))
+    }
+    val restoreJsonParams = restoreJsonParamsBuilder.build()
+
+    val repoToPushTo = params.getString("repoToPushTo")
+    val localWorkingCopyDirectory = params.getString("localWorkingCopyDirectory")
+
+    val git = setUpRepository(localWorkingCopyDirectory, repoToPushTo)
+
+    // Run ExportAnnotations.kt
+    pushLogger.info {"Beginning ExportAnnotations"}
+    ExportAnnotations.export(exportAnnotationsParams)
+
+    // Run RestoreJson.kt
+    if (restoreJson) {
+        pushLogger.info { "Beginning RestoreJson" }
+        RestoreJson.restore(restoreJsonParams)
+    }
+
+    // Push new annotations
+    pushUpdatedAnnotations(git)
+    pushLogger.info {"Done!"}
+}
+
+fun pushUpdatedAnnotations(git: Git) {
+
+    // Source: https://github.com/kittinunf/Kotlin-Playground/blob/master/buildSrc/src/main/java/com/github/kittinunf/gradle/ReleaseToMasterMergeTask.kt
+    class SshConfig(val sshKeyPath: String): TransportConfigCallback {
+        override fun configure(transport: Transport?) {
+            (transport as SshTransport).sshSessionFactory = object: JschConfigSessionFactory() {
+                override fun configure(hc: OpenSshConfig.Host?, session: Session?) {}
+                override fun createDefaultJSch(fs: FS?): JSch = super.createDefaultJSch(fs).apply {
+                    addIdentity(sshKeyPath)
+
+                }
+            }
+        }
+    }
+
+    val status = git.status().call()
+
+    // A clean status indicates no updates
+    if (!status.isClean()) {
+        val currentDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+        val formatter = SimpleDateFormat("yyyy-MM-dd-HH-mm-ss")
+
+        git.add().addFilepattern("data/").call()
+        git.commit()
+                .setMessage("includes annotations from ${currentDate.getTime()}").call()
+        git.tag().setName(formatter.format(currentDate.getTime())).call()
+        pushLogger.info {"Pushing updated data"}
+        git.push()
+                .setTransportConfigCallback(SshConfig("~/.ssh/id_rsa"))
+                .call()
+    } else {
+        pushLogger.info {"No updates - nothing will be pushed"}
+    }
+}
+
+fun setUpRepository(localWorkingCopyDirectory: String, repoToPushTo: String): Git {
+    /**
+     * Check if `localWorkingCopyDirectory` exists.
+     *   If it does: check the state
+     *   If not: clone `repoToPushTo` to the location specified by `localWorkingCopyDirectory`
+     */
+    if (File(localWorkingCopyDirectory).isDirectory()) {
+        val git = Git.open(File(localWorkingCopyDirectory))
+        val remoteUrl = git.getRepository()
+                .getConfig()
+                .getString("remote", "origin", "url")
+        // Does remote URL match `repoToPushTo`?
```

Comment:
This comment and the next three are good examples of the sort of comments which can be omitted, especially given the text of the exception messages.  For example, _Is the status clean?_ doesn't really add anything to `if (!status.isClean())`.  In general, you want the comments to explain not the _what_ of the code but rather the _why_ of the code.